### ISSUE TITLE
Unexpected token 'import' when using new ESLint config

### ISSUE THREAD

**Issue by EvHaus** [import-js/eslint-plugin-import#2536]
I'm looking into switching to [ESLint's new config system](https://eslint.org/blog/2022/08/new-config-system-part-2/) and its recommended ESM syntax. Everything is going well and the vast majority of my plugins are working just fine, except for `eslint-plugin-import` which fails with:

```
Oops! Something went wrong! :(

ESLint: 8.23.0

SyntaxError: Unexpected token 'import'
    at ESMLoader.moduleStrategy (node:internal/modules/esm/translators:117:18)
    at ESMLoader.moduleProvider (node:internal/modules/esm/loader:337:14)
```

To reproduce, create a repo with:

- `type: "module"` in `package.json`
- Create a `eslint.config.js` file with:
```js
import import from 'eslint-plugin-import';

export default {
	plugins: {
		// You don't even need this. Simply importing the plugin above, causes the error
		import
	},
};
```

Any clues as to what the problem might be? The stack trace isn't super helpful.

**Comment by ljharb** [comment]
(oof, I don't see any advantage to ever using native ESM in a config file)

While this plugin, via `resolve`, doesn't yet understand "exports" or `type: "module"` (which, ftr, is 100% unnecessary to use ESM and should be avoided regardless), this message comes from node.

I suspect the issue is that `import` is a restricted word, so you can't use it as a variable name.

**Comment by EvHaus** [comment]
> I suspect the issue is that import is a restricted word, so you can't use it as a variable name.

🤦 Duh, of course! Thanks for the quick response, and I apologize for opening up yet another ESM-related can-of-worms.

For those coming to here from the future -- the fix was:

```js
import importPlugin from 'eslint-plugin-import';

export default {
	plugins: {
		import: importPlugin
	},
};
```
